static/app/views/releases/drawer/foldSection.tsx

```diff
@@ -0,0 +1,205 @@
+      if (!element || navScrollMargin || hasAttemptedScroll.current) {
+        return;
+      }
+      // Prevent scrolling to element on rerenders
+      hasAttemptedScroll.current = true;
+
+      // scroll to element if it's the current section on page load
+      if (window.location.hash) {
+        const [, hash] = window.location.hash.split('#');
+        if (hash === sectionKey) {
+          if (isCollapsed) {
+            setIsCollapsed(false);
+          }
+
+          // Delay scrollIntoView to allow for layout changes to take place
+          setTimeout(() => element?.scrollIntoView(), 100);
+        }
+      }
+    },
+    [sectionKey, navScrollMargin, isCollapsed, setIsCollapsed]
+  );
+
+  // This controls disabling the InteractionStateLayer when hovering over action items. We don't
+  // want selecting an action to appear as though it'll fold/unfold the section.
+  const [isLayerEnabled, setIsLayerEnabled] = useState(true);
+
+  const toggleCollapse = useCallback(
+    (e: React.MouseEvent) => {
+      e.preventDefault(); // Prevent browser summary/details behaviour
+      window.getSelection()?.removeAllRanges(); // Prevent text selection on expand
+      setIsCollapsed(!isCollapsed);
+      // onToggleCollapse
+    },
+    [isCollapsed, setIsCollapsed]
+  );
+  const labelPrefix = isCollapsed ? t('View') : t('Collapse');
+  const labelSuffix = typeof title === 'string' ? title + t(' Section') : t('Section');
+
+  return (
+    <Fragment>
+      <Section
+        ref={mergeRefs([forwardedRef, scrollToSection])}
+        id={sectionKey}
+        scrollMargin={navScrollMargin ?? 0}
+        role="region"
+        className={className}
+      >
+        <SectionExpander
+          preventCollapse={preventCollapse}
+          onClick={preventCollapse ? e => e.preventDefault() : toggleCollapse}
+          role="button"
+          aria-label={`${labelPrefix} ${labelSuffix}`}
+          aria-expanded={!isCollapsed}
+        >
+          <InteractionStateLayer
+            hasSelectedBackground={false}
+            hidden={preventCollapse ? preventCollapse : !isLayerEnabled}
+          />
+          <TitleWithActions preventCollapse={preventCollapse}>
+            <TitleWrapper>{title}</TitleWrapper>
+            {!isCollapsed && (
+              <div
+                onClick={e => e.stopPropagation()}
+                onMouseEnter={() => setIsLayerEnabled(false)}
+                onMouseLeave={() => setIsLayerEnabled(true)}
+              >
+                {actions}
+              </div>
+            )}
+          </TitleWithActions>
+          <IconWrapper preventCollapse={preventCollapse}>
+            <IconChevron direction={isCollapsed ? 'down' : 'up'} size="xs" />
+          </IconWrapper>
+        </SectionExpander>
+        {isCollapsed ? null : (
+          <ErrorBoundary mini>
+            <Content>{children}</Content>
+          </ErrorBoundary>
+        )}
+      </Section>
+      <SectionDivider />
```

Comment:
ditto divider

Reply:
I'd probably prefer to keep this in here bc this is based on a component that does more stuff™ in issue details (ideally we could split that one into 2 components)